packages/mds-agency/request-handlers.ts

```diff
@@ -102,7 +103,9 @@ export const registerVehicle = async (req: AgencyApiRequest, res: AgencyApiRespo
 export const getVehicleById = async (req: AgencyApiRequest, res: AgencyApiResponse) => {
   const { device_id } = req.params
 
-  const { provider_id } = res.locals.scopes.includes('vehicles:read') ? req.query : res.locals
+  const { provider_id } = res.locals.scopes.includes('vehicles:read')
+    ? parseQuery()(req.query, 'provider_id')
```

Comment:
I think this would read a little better if you moved the `req.query` to the first parameter of `parseQuery` ... Then instead of just returning a function, return that function as an object property. This would result in the following call signature:
```ts
parseQuery(req.query).keys('provider _id')
```
Or
```ts
parseQuery(req.query, Number).keys('timestamp')
```
You could also set up the parameters of the returned `keys` function to require at least one key.

Reply:
SGTM